src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp

```diff
@@ -289,18 +298,33 @@ void FlightTaskManualAltitude::_rotateIntoHeadingFrame(Vector2f &v)
 
 void FlightTaskManualAltitude::_updateHeadingSetpoints()
 {
-	/* Yaw-lock depends on stick input. If not locked,
-	 * yaw_sp is set to NAN.
-	 * TODO: add yawspeed to get threshold.*/
-	if (fabsf(_yawspeed_setpoint) > FLT_EPSILON) {
-		// no fixed heading when rotating around yaw by stick
-		_yaw_setpoint = NAN;
+	_isYawInput()
+	? _unlockYaw()
+	: _lockYaw();
+}
 
-	} else {
-		// hold the current heading when no more rotation commanded
-		if (!PX4_ISFINITE(_yaw_setpoint)) {
-			_yaw_setpoint = _yaw;
-		}
+bool FlightTaskManualAltitude::_isYawInput()
+{
+	/*
+	 * A threshold larger than FLT_EPSILON is required because the
+	 * _yawspeed_setpoint comes from an IIR filter and takes too much
+	 * time to reach zero.
+	 */
+	return fabsf(_yawspeed_setpoint) > 0.001f;
+}
+
+void FlightTaskManualAltitude::_unlockYaw()
+{
+	// no fixed heading when rotating around yaw by stick
+	_yaw_setpoint = NAN;
+}
+
+void FlightTaskManualAltitude::_lockYaw()
+{
+	// hold the current heading when no more rotation commanded
+	if (!PX4_ISFINITE(_yaw_setpoint)) {
+		_yaw_setpoint = _yaw;
+		_yawspeed_setpoint = 0.f;
```

Comment:
```suggestion
		_yawspeed_setpoint = NAN;
```
The heading is locked anyways. Adding feedforward is not necessary.

Reply:
was changed